test/yki/handler/participant_test.clj

```diff
@@ -70,13 +71,19 @@
       (is (= registration-state "CANCELLED")))))
 
 (deftest exam-session-participant-delete-paid-test
-  (testing "delete exam session participant should set registration state to PAID_AND_CANCELLED when registration is paid"
-    (base/insert-base-data)
-    (base/insert-registrations "COMPLETED")
-    (let [registration-id    (:id (base/select-one "SELECT id from registration"))
-          exam-session-route (get-exam-session-route)
-          request            (mock/request :delete (str exam-session-route "/registration/" registration-id))
-          response           (base/send-request-with-tx request)
-          registration-state (:state (base/select-one (str "SELECT state from registration where id=" registration-id)))]
-      (is (= (:status response) 200))
-      (is (= registration-state "PAID_AND_CANCELLED")))))
+  (base/insert-base-data)
+  (base/insert-registrations "COMPLETED")
+  (let [registration-id        (:id (base/select-one "SELECT id from registration"))
+        exam-session-route     (get-exam-session-route)
+        request                (mock/request :delete (str exam-session-route "/registration/" registration-id))
+        get-registration-state #(:state (base/select-one (str "SELECT state from registration where id=" registration-id)))]
+    (testing "cancelling paid registration should fail if cancelling user is not OPH admin"
+      (with-redefs [auth/oph-admin-access (constantly false)]
+        (let [response (base/send-request-with-tx request)]
+          (is (= (:status response) 404))
```

Comment:
Olisko parempi palauttaa 403 ja vaihtaa myös handlerin tasolle uusi virheviesti, nyt `Registration not found` ?

Reply:
Vaihdan palauttamaan statuksen 400. Tässä nyt tietysti menee pari eri virheskenaariota saman statuskoodin alle (ilmoittautumista ei löytynyt, ilmoittautumisen tila ei vastaa odotettua), mutta ehkäpä se ei haittaa kun vain poikkeustapauksissa näitä pitäisi sattua.